server/fleet/activities.go

```diff
@@ -802,3 +803,82 @@ func (a ActivityTypeEditedMacosProfile) Documentation() (activity, details, deta
   "team_name": "Workstations"
 }`
 }
+
+// LogRoleChangeActivities logs activities for each role change, globally and one for each change in teams.
+func LogRoleChangeActivities(ctx context.Context, ds Datastore, adminUser *User, oldGlobalRole *string, oldTeamRoles []UserTeam, user *User) error {
+	if user.GlobalRole != nil && (oldGlobalRole == nil || *oldGlobalRole != *user.GlobalRole) {
+		if err := ds.NewActivity(
+			ctx,
+			adminUser,
+			ActivityTypeChangedUserGlobalRole{
+				UserID:    user.ID,
+				UserName:  user.Name,
+				UserEmail: user.Email,
+				Role:      *user.GlobalRole,
+			},
+		); err != nil {
+			return err
+		}
+	}
+	if user.GlobalRole == nil && oldGlobalRole != nil {
+		if err := ds.NewActivity(
+			ctx,
+			adminUser,
+			ActivityTypeDeletedUserGlobalRole{
+				UserID:    user.ID,
+				UserName:  user.Name,
+				UserEmail: user.Email,
+				OldRole:   *oldGlobalRole,
+			},
+		); err != nil {
+			return err
+		}
+	}
+	oldTeamsLookup := make(map[uint]UserTeam, len(oldTeamRoles))
+	for _, t := range oldTeamRoles {
+		oldTeamsLookup[t.ID] = t
+	}
+
+	newTeamLookup := make(map[uint]struct{}, len(user.Teams))
```

Comment:
nit: newTeamLookup --> newTeamsLookup

Reply:
Will do. Also, this code was moved an exported (I reused it.).

Reply:
Actually my bad, I forgot to remove the now exported method, see 23f4b4a06. (Method `logRoleChangeActivities` is now exported in the `fleet` package as `LogRoleChangeActivities`.)